src/app/games/quebec-castles/QuebecCastlesRules.ts

```diff
@@ -0,0 +1,686 @@
+import { MGPFallible, MGPOptional, MGPValidation, Utils } from '@everyboard/lib';
+
+import { GameNode } from 'src/app/jscaip/AI/GameNode';
+import { QuebecCastlesDrop, QuebecCastlesMove, QuebecCastlesTranslation } from './QuebecCastlesMove';
+import { QuebecCastlesState } from './QuebecCastlesState';
+import { GameStatus } from 'src/app/jscaip/GameStatus';
+import { ConfigurableRules } from 'src/app/jscaip/Rules';
+import { Table, TableUtils } from 'src/app/jscaip/TableUtils';
+import { BooleanConfig, EnumConfig, NumberConfig, RulesConfigDescription, RulesConfigDescriptionLocalizable } from 'src/app/components/wrapper-components/rules-configuration/RulesConfigDescription';
+import { MGPValidators } from 'src/app/utils/MGPValidator';
+import { Coord, CoordFailure } from 'src/app/jscaip/Coord';
+import { Player, PlayerOrNone } from 'src/app/jscaip/Player';
+import { PlayerMap } from 'src/app/jscaip/PlayerMap';
+import { RulesFailure } from 'src/app/jscaip/RulesFailure';
+import { MoveCoordToCoord } from 'src/app/jscaip/MoveCoordToCoord';
+import { DirectionFailure } from 'src/app/jscaip/Direction';
+import { Ordinal } from 'src/app/jscaip/Ordinal';
+import { Localized } from 'src/app/utils/LocaleUtils';
+
+export class QuebecCastlesFailure {
+
+    public static readonly INVALID_INVADER_DISTANCE: (distance: number) => string = (distance: number) => $localize`Move distance must be 2 for invader, not ${ distance }`;
+
+    public static readonly INVALID_DEFENDER_DISTANCE: (distance: number) => string = (distance: number) => $localize`Move distance must be 1 for defender, not ${ distance }`;
+
+    public static readonly MUST_DROP_IN_YOUR_TERRITORY: Localized = () => $localize`Must drop in your own territory`;
+
+    public static readonly CANNOT_DROP_IN_MOVE_PHASE: Localized = () => $localize`Cannot drop in move phase`;
+
+    public static readonly CANNOT_MOVE_IN_DROP_PHASE: Localized = () => $localize`Cannot drop in move phase`;
+
+    public static readonly MUST_DROP_ALL_YOUR_PIECES: Localized = () => $localize`Must drop all your pieces`;
+
+    public static readonly MUST_DROP_ALL_YOUR_REMAINING_PIECES: Localized = () => $localize`Must drop all your remaining pieces, not more not less`;
+
+    public static readonly CANNOT_DROP_THAT_MUCH: Localized = () => $localize`Cannot drop that much pieces`;
+
+    public static readonly MUST_DROP_ONE_BY_ONE: Localized = () => $localize`Must drop pieces one by one`;
+
+    public static readonly CANNOT_LAND_IN_YOUR_THRONE: Localized = () => $localize`You cannot land on your throne`;
+
+    public static readonly PLACE_ONLY_ONE_THRONE: Localized = () => $localize`You must only place your throne`;
```

Comment:
curieux de voir quand ce message apparait, j'ai du mal à voir !
Sinon, "you must only place your throne" serait peut-être plus approprié, en fonction